dangerzone/isolation_provider/qubes.py

```diff
@@ -38,6 +39,8 @@ def read_bytes(p: subprocess.Popen, buff_size: int) -> bytes:
 def read_int(p: subprocess.Popen) -> int:
     """Read 2 bytes from stdout, and decode them as int."""
     untrusted_int = p.stdout.read(2)  # type: ignore [union-attr]
+    if untrusted_int == b"":
+        raise ValueError("Nothing read from stdout (expected an integer)")
```

Comment:
Argh, that was stupid of me, when introducing these helpers. You're right for handling this case, but it should be handled in a uniform way across the code:

1. We should call the `read_bytes` function, instead of reading directly from `p.stdout`.
2. The `read_bytes` function needs an `exact=True` argument. If `False`, we can just return the read bytes. If `True` though, and the number of bytes is less than the specified one, we should raise an exception. This way, we can also avoid corrupted images in case of EOF.

Note that ideally, we may have to retry `read()`, if we don't want to rely on the `BufferedReader` type. See https://stackoverflow.com/questions/7383464/does-pythons-read-always-return-requested-size-except-at-eof

This is something I can help with in a separate PR.

Reply:
I have implemented this in the client-side timeout PR, since it ended up being needed there. We don't need to get rid of this line now. Once we have a merge conflict, we can tackle it accordingly.